harness/determined/common/experimental/determined.py

```diff
@@ -235,6 +237,19 @@ def get_checkpoint(self, uuid: str) -> checkpoint.Checkpoint:
         resp = bindings.get_GetCheckpoint(self._session, checkpointUuid=uuid)
         return checkpoint.Checkpoint._from_bindings(resp.checkpoint, self._session)
 
+    def get_workspace(self, workspace_id: int) -> workspace.Workspace:
+        resp = bindings.get_GetWorkspace(self._session, id=workspace_id)
+        return workspace.Workspace._from_bindings(resp.workspace, self._session)
+
+    def list_workspaces(self) -> List[workspace.Workspace]:
+        def get_with_offset(offset: int) -> bindings.v1GetWorkspacesResponse:
+            return bindings.get_GetWorkspaces(self._session, offset=offset)
+
+        iter_workspaces = itertools.chain.from_iterable(
```

Comment:
nit: what's the benefit of `itertools` over a vanilla for loop? the internet says it's faster, so i get that, but it's less readable IMO, and requires an extra dependency.